srcs/app/server/Server.cpp

```diff
@@ -5,16 +5,21 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <cerrno>
 #include <cstring>
+#include <iostream>
+#include <stdexcept>
 
 Server::Server(const ServerConfig &settings)
 	: settings_(settings), listen_sd_(-1) {
 }
 
-bool	Server::BindListeningSocket() {
-	// TODO(any) Error checking
+void	Server::BindListeningSocket() {
 	listen_sd_ = socket(AF_INET, SOCK_STREAM, 0);
-	fcntl(listen_sd_, F_SETFL, O_NONBLOCK);
+	if (listen_sd_ < 0)
+		throw std::runtime_error(std::strerror(errno));
```

Comment:
just one stupid format comment here, do you like the idea of using braces with single `if` ? 

```suggestion
	if (listen_sd_ < 0) {
	    throw std::runtime_error(std::strerror(errno));
	}
```
I normally preffer this approach because improves readability and prevent mistakes (losing identation for example), just wanted to point this out

Reply:
I like this :+1: , done.